### PR TITLE
work around for issue with unhandled exceptions . Issue #33

### PR DESCRIPTION
**Pull request by proind**
workaround for Issue #33  Issue With Exceptions. 
it doesn't actually have to do with logging.
see comment in code for details

### CODE REVIEW

File: python/ipythonEmbed.py [hunk 1]
```diff
@@ -145,7 +145,15 @@ def start(argv=None):
...
-
+                """
+                 Starting with  ipython 4.2.0 whenever certain exceptions are thrown, there is a call to get_terminal_size().
+                 in that function , in case environment variables for "COLUMNS" and "LINES" are not defined there is a call
+                 to sys.__stdout__.fileno()   in order to get a handle to the current terminal. IDAPythonStdOut doesn't have an attribute fileno
+                 so the call fails , and the kernel dies. the right way to solve it, is add AttributeError to the try/except in get_terminal_size.
+                 a work around is to add this 2 environment variables
+                """
+                os.environ["COLUMNS"] = "70"
+                os.environ["LINES"]   = "70"
```
Comment (tmr232) [inline python/ipythonEmbed.py]: Maybe change to something more standard - 80x24 (Python's default values) or 75 columns (IPython's max columns)?

**Comment by proind** [comment]
Sure, I'll change to 75

**Comment by tmr232** [comment]
This works great! Awesome!
I only have that issue with the numbers you chose.

I am also submitting an issue to https://github.com/ipython/ipython , they should not try and get the terminal size for things that are not terminals...

**Comment by tmr232** [comment]
The discussion on https://github.com/ipython/ipython/issues/10040 is interesting. We might be able to find a cleaner solution.

**Comment by proind** [comment]
I think the best solution is this 
[proind/backports.shutil_get_terminal_size/commit](https://github.com/proind/backports.shutil_get_terminal_size/commit/8b956c8f587f5e8a409329bdc4d97619d9b30860)

this way if fileno doesn't exist , the function will just return the default values.

**Comment by proind** [comment]
I changed the numbers. 
Also  did a pull request to get_terminal_size: [backports.shutil_get_terminal_size/pull/12](https://github.com/chrippa/backports.shutil_get_terminal_size/pull/12l)

**Comment by james91b** [comment]
Another solution is to add a dummy fileno function to IDAPythonStdOut. Maybe we can request they add in for a future release of IDAPython or we could just monkey patch it in.

``` python
#When__stdout__ is IDAPythonStdOut 
sys.__stdout__.fileno = lambda: None
```

The environment variables solution may be the cleanest though. Thoughts?

**Comment by proind** [comment]
I like the environment variables solution because it depends less on changes in other packages

**Comment by tmr232** [comment]
I agree - the environment variables are seem like the cleanest solution.

**Comment by tmr232** [comment]
So what is keeping us from going ahead with a solution?

**Comment by proind** [comment]
unless there is something else that needs to be changed , I think you can merge, no ?

**Comment by james91b** [comment]
Everything looks good. I'll merge it. Thanks.

**Pull request merged**